kpt-functions/replacement/function.go

```diff
@@ -97,92 +97,67 @@ type Function struct {
 	Config *FunctionConfig
 }
 
-func NewFunction(cfg *FunctionConfig) (*Function, error) {
-	for _, r := range cfg.Replacements {
-		if r.Source == nil {
-			return nil, fmt.Errorf("`from` must be specified in one replacement")
-		}
-		if r.Target == nil {
-			return nil, fmt.Errorf("`to` must be specified in one replacement")
-		}
-		count := 0
-		if r.Source.ObjRef != nil {
-			count += 1
-		}
-		if r.Source.Value != "" {
-			count += 1
-		}
-		if r.Source.MultiRef != nil {
-			count += 1
-		}
-		if count > 1 {
-			return nil, fmt.Errorf("only one of fieldref and value is allowed in one replacement")
-		}
+func (r Replacement) Validate() error {
+	if r.Source == nil {
+		return fmt.Errorf("`from` must be specified in one replacement")
 	}
-
-	fn := Function{Config: cfg}
-	return &fn, nil
+	if r.Target == nil {
+		return fmt.Errorf("`to` must be specified in one replacement")
+	}
+	count := 0
+	if r.Source.ObjRef != nil {
+		count += 1
+	}
+	if r.Source.Value != "" {
+		count += 1
+	}
+	if r.Source.MultiRef != nil {
+		count += 1
+	}
+	// @aodinokov, Should it be equal 1?
```

Comment:
I think we should report error when more than 1 contradicting fields are set.
`> 1` is right